### PR TITLE
Mutation data counts

### PR DESCRIPTION
**Pull request by 7xuanlu**
PR for https://github.com/cBioPortal/cbioportal/issues/10357

Do mutation data counts by gene specific
Option 1: based on whether the sample is mutated or not
Mutated (the samples with one or more mutations in the gene)
Wild Type (the samples without mutations in the gene)
NA (the samples that were not profiled)

Option 2: based on mutation type
Missense Mutation, Frame Shift Del, Frame Shift Ins, etc...

Implement new MutationDataFilters to filter by above groups

### CODE REVIEW

File: src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java [hunk 1]
```diff
@@ -134,4 +135,10 @@ public MutationCountByPosition getMutationCountByPosition(Integer entrezGeneId,
         return mutationMapper.getMutationCountByPosition(entrezGeneId, proteinPosStart, proteinPosEnd);
     }
 
+    @Override
+    public GenomicDataCountItem getMutationCountsByType(List<String> molecularProfileIds, List<String> sampleIds, 
+                                                       List<Integer> entrezGeneIds, String profileType) {
+        return mutationMapper.getMutationCountsByType(molecularProfileIds, sampleIds, entrezGeneIds,
```
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java:141]: @Override
public MutationMeta fetchMetaMutationsInMolecularProfile(String molecularProfileId, List<String> sampleIds,
                                                         List<Integer> entrezGeneIds) {

    return mutationMapper.getMetaMutationsBySampleIds(molecularProfileId, sampleIds, entrezGeneIds, null);
}
    
I adopted Ersin's approach regarding the snpOnly parameter. It aligns with my understanding. Would you like me to modify his code accordingly?

File: src/main/java/org/cbioportal/service/impl/MutationServiceImpl.java [hunk 2]
```diff
@@ -1,9 +1,6 @@
 package org.cbioportal.service.impl;
 
-import org.cbioportal.model.GeneFilterQuery;
-import org.cbioportal.model.MolecularProfile;
-import org.cbioportal.model.Mutation;
-import org.cbioportal.model.MutationCountByPosition;
+import org.cbioportal.model.*;
```
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/service/impl/MutationServiceImpl.java:3]: It shows no * imports

[the remark this answers is not in the document]
Comment (JREastonMarks) [reply src/main/java/org/cbioportal/service/impl/MutationServiceImpl.java:3]: I think it still shows import org.cbioportal.model.*;

File: src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (JREastonMarks) [reply src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java:141]: If setting the variable has no changes in the SQL or results being returned then it shouldn't be there.

File: src/main/java/org/cbioportal/service/util/MolecularProfileUtil.java [hunk 3]
```diff
@@ -44,18 +61,13 @@ public List<MolecularProfileCaseIdentifier> getFilteredMolecularProfileCaseIdent
         for (int i = 0; i < studyIds.size(); i++) {
```
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/service/util/MolecularProfileUtil.java:61]: Renamed to studyIdIdx

File: src/main/java/org/cbioportal/service/impl/MutationServiceImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/service/impl/MutationServiceImpl.java:3]: Got it. Resolved.

File: src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java:141]: `Caused by: org.apache.ibatis.binding.BindingException: Parameter 'snpOnly' not found.`
Above appears when we remove **snpOnly** from **MutationMapper.java** and **MutationMyBatisRepository.java**. MutationMapper.xml requires us to pass **snpOnly** to the function even though it's fixed at null

File: src/main/java/org/cbioportal/web/parameter/Categorization.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package org.cbioportal.web.parameter;
+
+public enum Categorization {
```
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/web/parameter/Categorization.java:3]: deleted, the naming is a bad idea

File: src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (7xuanlu) [reply src/main/java/org/cbioportal/persistence/mybatis/MutationMyBatisRepository.java:141]: added

**Review by JREastonMarks — APPROVED** [review]
LGTM

**Pull request merged**